types/zeit__next-typescript/index.d.ts

```diff
@@ -4,11 +4,9 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.8
 
-import { ServerConfig } from 'next';
-
 declare function withTypescript(
     /** @default {} */
-    nextConfig?: ServerConfig
-): ServerConfig;
+    nextConfig?: { [key: string]: any }
+): { [key: string]: any };
```

Comment:
The *actual* TS types for this plugin for Next.js 9.0 is indeed `{ [key: string]: any }`, as per [this file](https://github.com/zeit/next.js/blob/7c80febcf796c91bb776ea3b841cab18df221ff4/packages/next/next-server/server/config.ts#L12)